src/util/interval_template.h

```diff
@@ -143,6 +143,47 @@ template<class T> class interval_templatet
     }
   }
 
+  void make_bottom()
+  {
+    lower_set = upper_set = true;
+    upper = T();
+    lower = upper + 1;
+  }
+
+  void make_less_than_eq(interval_templatet &i)
+  {
+    if(upper_set && i.upper_set)
```

Comment:
If my own upper bound is not set but i's is, shouldn't it now become set, and the same for i's lower bound inheriting mine?